### ISSUE TITLE
k8s-alpha create assumes that public ssh key contains filename

### ISSUE THREAD

**Issue by kengu** [linode/linode-cli#92]
**Note**: This error is only reproduced on mac.

The k8s-alpha plugin assumes that the public ssh key contains the filename, which I guess is not stable across different OS implementations. On mac, the default comment `user@hostname`, which does not match the key filename which the script checks for. As a result, the create command generates the warning "Your ssh private key must be added to your ssh-agent." even if you have added the key. The workaround I used was to generate a key that the script accepted was

```
> ssh-keygen -b 4096 -t rsa -C "id_rsa"`
> ssh-add -K ~/.ssh/id_rsa
> linode-cli k8s-alpha create k8s --ssh-public-key ~/.ssh/id_rsa.pub
```

See [Generating a new SSH key and adding it to the ssh-agent](https://help.github.com/articles/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent/#platform-mac)

**Comment by asauber** [comment]
Hi @kengu, thanks for the report

The plugin doesn't assume that the key data contain the filename, but it does assume that `ssh-add -l` output filepaths.

For example here's that output on my system:
```
asauber@linaret2 linode-cloud-controller-manager # ssh-add -l
2048 SHA256:Qcl/i4Yd9E7vXQbiJ98rFTHssJ1TVGKyaRYsjstWypg /Users/asauber/.ssh/clidemo (RSA)
4096 SHA256:Pv0U9hFqENpF5cW0dqrsr5zA7XYgHaNtYPRpzEMe0Hg /Users/asauber/.ssh/k8sdemo (RSA)
2048 SHA256:OcziYNj7HWb1yZINdJxXy59dTjDqZ1IwYdul6ddTeVc /Users/asauber/.ssh/id_rsa (RSA)
4096 SHA256:DpwXaeZ25N5D9CaFLy4aa5pumxvOVhNZ5qW3xI6wOyY /Users/asauber/.ssh/testkey (RSA)
```

**Comment by asauber** [comment]
Does yours only output the comment where mine outputs the filepath? Maybe we're running different versions of `ssh-add`. I tested that the "filepath output" behavior was consistent on a few Linux distributions before using it.

There must be some differences between our OS configurations or our version of `ssh-add`

Unfortunately `ssh-add` doesn't have a version flag, so I'm skimming through `strings $(which ssh-add)` at the moment.

**Comment by kengu** [comment]
I agree, I think the reason is differences in builds of ssh-add for Mac and Linux. As far as I can see, the output from `ssh-add -l` [is undocumented](https://linux.die.net/man/1/ssh-add) so I guess default implementation off ssh-add made by Apple is different than OpenSSH`s? Since output is not documented, the plugin should probably rely on some other mechanism to validate that the correct key is supplied?

**Comment by kengu** [comment]
I've done some research into openssh, which according to [wikipedia](https://en.wikipedia.org/wiki/Comparison_of_SSH_clients) is included in OSX. Both [openssh](https://anongit.mindrot.org/openssh.git/tree/ssh-add.c#n442) and [openssh-portable](https://github.com/openssh/openssh-portable/blob/42c5ec4b97b6a1bae70f323952d0646af16ce710/ssh-add.c#L441) (used by macos) does not output filepath. I guess the filepath is added to the comment segment of the key by ssh-keygen?

**Comment by asauber** [comment]
Hi @kengu, I believe we addressed this concern with #110

**Comment by asauber** [comment]
With another report of this issue, I was able to track down the exact cause.

https://github.com/openssh/openssh-portable/blob/master/ssh-add.c#L275
If a key added to an agent doesn't have the comment, then the comment that the agent stores is the filepath of the key.

My keys are in RSA format, which don't have comments. The OPENSSH format is popular now, which always have comments which default to user@hostname.

So, my assumption in the plugin that I can grab the filename from `ssh-agent -l` is definitely not true.

I'll work on a fix.
